internal/exec/util/unit.go

```diff
@@ -110,8 +113,58 @@ func (ut Util) MaskUnit(unit types.Unit) (string, error) {
 	return filepath.Join("/", SystemdUnitsPath(), unit.Name), nil
 }
 
+var builder = strings.Builder{}
+
+// isRequiredSystemdVersionExists checks if the version of systemd is greater
+// than or equal to 240 for leveraging this (https://github.com/systemd/systemd/pull/9901) feature merged in the v240 branch.
+func isRequiredSystemdVersionExists() (bool, error) {
```

Comment:
I think we can drop this function entirely and just assume the systemd patch is there. Currently, the master branch can and does assume newer components. See for example: https://github.com/coreos/ignition/pull/846.

If we need to backport this to RHCOS, we can ask the systemd folks to backport the corresponding patch too.

Reply:
You think it would be that easy to get it backported? I assumed that would be hard to get pulled in in a reasonable time frame.

Reply:
Is there a request to have this backported soon?

If we really need to, my suggestion is to make it a compile-time switch instead we can turn on/off from the spec file.

Reply:
I agree with @jlebon here. Additionally, backporting things throws a wrench into feature auto-detection by version probing (as you end up with the same old versions which both do and do not support the new feature).

Reply:
I don't currently know of an RFE to have it backported. I guess I can look into opening one.

Regarding a compile time flag, I guess we can do that, though another option comes to mind. We could create a version of this PR that unconditionally adopts the new format and see if it breaks old systemd. Since enabling instantiated units is already broken as long as systemd doesn't crash or somehow not enable other units then the functionality will be equivalent to what we already have on systemd < 240.

I'd say if we do encounter systemd < 240 we should at least print a warning with a good message (possibly linking to https://github.com/coreos/ignition/issues/586)

Reply:
Specifically re.

> I don't currently know of an RFE to have it backported. I guess I can look into opening one.

:+1:

For example, Colin just opened one for having https://github.com/systemd/systemd/pull/14154 backported: https://bugzilla.redhat.com/show_bug.cgi?id=1812894

Reply:
here's the BZ with a request for backport: https://bugzilla.redhat.com/show_bug.cgi?id=1812972

any more thoughts on my suggestion in https://github.com/coreos/ignition/pull/934#discussion_r391632108 ?

Reply:
> Regarding a compile time flag, I guess we can do that, though another option comes to mind. We could create a version of this PR that unconditionally adopts the new format and see if it breaks old systemd. Since enabling instantiated units is already broken as long as systemd doesn't crash or somehow not enable other units then the functionality will be equivalent to what we already have on systemd < 240.

Hmm yeah, I think that makes sense overall.

Reply:
I've performed some testing around this feature to assess the exact impact on the RHCOS system. 
So here's some observation when I ran the following example on RHCOS(`rhcos-44.81`)
**Systemd's version**
```
[core@localhost system-preset]$ systemctl --version
systemd 239
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=legacy
```
**Preset file**
```
[core@localhost]$ cat /etc/systemd/system-preset/20-ignition.preset 
enable echo@bar.service 
enable echo@foo.service
disable sshd.service
```
```
[core@localhost]$ sudo systemctl preset-all
Removed /etc/systemd/system/network-online.target.wants/NetworkManager-wait-online.service.
Removed /etc/systemd/system/vmtoolsd.service.requires/vgauthd.service.
Created symlink /etc/systemd/system/network-online.target.wants/NetworkManager-wait-online.service → /usr/lib/systemd/system/NetworkManager-wait-online.service.
Created symlink /etc/systemd/system/vmtoolsd.service.requires/vgauthd.service → /usr/lib/systemd/system/vgauthd.service
```
If you see the o/p after `systemctl preset-all` command, it's simply ignoring the instantiated unit entries listed in the `20-ignition.preset` file. And one good thing over here is - it's not breaking the system at all so we can safely include this feature in the latest release of `FCOS` and later backport it to `RHCOS` if feasible.

Reply:
Thanks @sohankunkerkar - that describes the current behavior. What happens if you put the output of this PR into the preset file?:

```
enable echo@.service bar foo
disable sshd.service
```

Reply:
I have updated my previous explanation. I did test that use-case and the o/p remain the same.

Reply:
OK so that means the new syntax won't cause any errors in old systemd. That's good news.